# AtmoSync
## 01 - Data Collection and Dataset Understanding

### Objective

The purpose of this notebook is to load the raw microclimate dataset and understand its structure before performing any cleaning or analysis.

### Tasks

1. Load the dataset.
2. Inspect the number of rows and columns.
3. Examine column names.
4. Check data types.
5. Check missing values.
6. Check duplicate records.
7. Examine basic statistics.
8. Identify potential data-quality problems.

In [1]:
import pandas as pd
import numpy as np
from pathlib import Path

# ---------------------------------------------------------
# Define project paths
# ---------------------------------------------------------

# Project root directory
PROJECT_ROOT = Path.cwd().parent

# Location of the raw dataset
DATA_PATH = (
    PROJECT_ROOT
    / "dataset"
    / "raw"
    / "Microclimate_dataset.csv"
)

# ---------------------------------------------------------
# Load dataset
# ---------------------------------------------------------

df = pd.read_csv(DATA_PATH)

print("Dataset loaded successfully.")
print(f"Rows: {df.shape[0]}")
print(f"Columns: {df.shape[1]}")

Dataset loaded successfully.
Rows: 8640
Columns: 12


In [2]:
# Display the first five records

df.head()

,cdTimestamp,Location_ID,Latitude,Longitude,LULC_Type,Temperature,Humidity,Wind_Speed,Solar_Radiation,Precipitation,Air_Pressure,NDVI
0,2022-01-01 00:00:00,Station_1,42.920785,-70.442076,Agricultural,-1.45,30.99,1.03,747.73,9,980.36,0.87
1,2022-01-01 00:05:00,Station_1,45.878954,-88.553180,Forest,24.93,58.33,9.05,297.29,0,1020.30,0.56
2,2022-01-01 00:10:00,Station_1,32.147619,-91.387175,Agricultural,20.60,74.46,8.63,549.58,2,1024.56,0.52
3,2022-01-01 00:15:00,Station_1,26.725934,-114.654620,Barren,3.09,65.30,7.90,755.51,10,1043.17,0.46
4,2022-01-01 00:20:00,Station_1,40.877759,-92.345482,Barren,17.44,44.88,11.49,979.98,1,998.33,0.47


In [3]:
# Display all column names

df.columns.tolist()

['cdTimestamp',
 'Location_ID',
 'Latitude',
 'Longitude',
 'LULC_Type',
 'Temperature',
 'Humidity',
 'Wind_Speed',
 'Solar_Radiation',
 'Precipitation',
 'Air_Pressure',
 'NDVI']

In [4]:
# Check data types

df.dtypes

cdTimestamp            str
Location_ID            str
Latitude           float64
Longitude          float64
LULC_Type              str
Temperature        float64
Humidity           float64
Wind_Speed         float64
Solar_Radiation    float64
Precipitation        int64
Air_Pressure       float64
NDVI                   str
dtype: object

In [5]:
# Check missing values

missing_values = df.isnull().sum()

missing_values

cdTimestamp        0
Location_ID        0
Latitude           0
Longitude          0
LULC_Type          0
Temperature        0
Humidity           0
Wind_Speed         0
Solar_Radiation    0
Precipitation      0
Air_Pressure       0
NDVI               0
dtype: int64

In [6]:
# Check duplicate records

duplicate_count = df.duplicated().sum()

print("Duplicate records:", duplicate_count)

Duplicate records: 0


In [7]:
# Statistical summary

df.describe(include="all")

,cdTimestamp,Location_ID,Latitude,Longitude,LULC_Type,Temperature,Humidity,Wind_Speed,Solar_Radiation,Precipitation,Air_Pressure,NDVI
count,8640,8640,8640.000000,8640.000000,8640,8640.000000,8640.000000,8640.000000,8640.000000,8640.000000,8640.000000,8640
unique,288,30,NaN,NaN,5,NaN,NaN,NaN,NaN,NaN,NaN,82
top,2022-01-01 00:00:00,Station_1,NaN,NaN,Urban,NaN,NaN,NaN,NaN,NaN,NaN,0.54
freq,30,288,NaN,NaN,1775,NaN,NaN,NaN,NaN,NaN,NaN,132
mean,NaN,NaN,37.564648,-92.608464,NaN,12.514738,60.303934,7.562444,547.501514,4.994792,1015.173694,NaN
std,NaN,NaN,7.199839,18.747068,NaN,13.055157,17.446686,4.318258,261.446596,3.146476,20.309088,NaN
min,NaN,NaN,25.000662,-124.999231,NaN,-10.000000,30.010000,0.000000,100.050000,0.000000,980.030000,NaN
25%,NaN,NaN,31.366795,-108.487303,NaN,1.225000,44.997500,3.860000,321.130000,2.000000,997.610000,NaN
50%,NaN,NaN,37.714368,-92.777086,NaN,12.420000,60.530000,7.610000,548.460000,5.000000,1015.190000,NaN
75%,NaN,NaN,43.703949,-76.515592,NaN,23.840000,75.510000,11.310000,770.282500,8.000000,1032.805000,NaN


## Initial Data Quality Observations

The dataset contains 8,640 observations.

The initial inspection should identify:

- missing values
- duplicate records
- incorrect data types
- invalid values
- potential outliers
- categorical variables

One known data-quality issue is that the NDVI column contains an invalid text value and therefore requires conversion to numeric format during the cleaning stage.